src/pg/pg-store.ts

```diff
@@ -77,6 +78,9 @@ export class PgStore extends BasePgStore {
             }
           }
         }
+        const block_height = event.block_identifier.index;
+        await this.rollBackInscriptionCount({ block_height });
+        logger.info(`PgStore rollback inscription counts for block ${block_height}`);
```

Comment:
Instead of figuring out rollbacks and inserts within these transactions, you should keep a list of affected block heights and recompute totals after these transactions are settled (similar to what I currently do with `normalizeInscriptionLocations`)